### PR TITLE
Fixed omission in USE_SESSION_FOR_NEXT

### PR DESCRIPTION
**Pull request by HoustonFortney**
My last PR #330 had an error on my part. I didn't realize that fresh logins were handled separately. I updated that function to use the session as well and added a test for this.

Sorry about that.

I did a ctrl-F audit of `login_manager.py` and it looks I have caught all instances of `login_url` so I think we are good now.

### CODE REVIEW

File: flask_login/login_manager.py [hunk 1]
```diff
@@ -274,7 +274,15 @@ def needs_refresh(self):
             flash(self.needs_refresh_message,
                   category=self.needs_refresh_message_category)
 
-        return redirect(login_url(self.refresh_view, request.url))
+        config = current_app.config
+        if config.get('USE_SESSION_FOR_NEXT', USE_SESSION_FOR_NEXT):
+            session['next'] = make_next_param(
```
Comment (alanhamlett) [inline flask_login/login_manager.py]: Another way to write this:
```
login_url = expand_login_view(self.refresh_view)
session['next'] = make_next_param(login_url, request.url)
```
It's easier to read for me. Any thoughts?

**Comment by alanhamlett** [comment]
Thanks, good catch. Also looked through the code for usages of `login_url` and `make_next_param` and your test covers them.

**Comment by HoustonFortney** [comment]
@alanhamlett done. I agree it is more readable.

**Comment by alanhamlett** [comment]
Thanks @HoustonFortney, also if you have time I would appreciate thoughts on #335.

**Pull request merged**